### ISSUE TITLE
useValidAutocomplete ignores React's camelCase `autoComplete` (only matches lowercase `autocomplete`)

### ISSUE THREAD

**Issue by hudson155** [biomejs/biome#10739]
### Environment information

Biome `2.5.0` (latest). The relevant code on `main` is unchanged, so it reproduces there too.

```
Biome version: 2.5.0
```

### Rule

`lint/a11y/useValidAutocomplete`

### What happened

`useValidAutocomplete` resolves the attribute with an exact, case-sensitive lookup for the literal name `autocomplete`:

```rust
// crates/biome_js_analyze/src/lint/a11y/use_valid_autocomplete.rs
let autocomplete_attribute = node.attributes().find_by_name("autocomplete")?;
```

`JsxAttributeList::find_by_name` compares the raw source text with no case normalization:

```rust
// crates/biome_js_syntax/src/jsx_ext.rs
name.value_token().ok()?.text_trimmed() == name_to_lookup
```

In React/JSX the attribute is spelled `autoComplete` (camelCase). `"autoComplete" != "autocomplete"`, so the lookup returns `None` and the rule returns before validating anything. As a result **the rule never fires on idiomatic React/TSX** — it only catches the lowercase `autocomplete` spelling, which React itself warns about at runtime.

This diverges from the upstream `eslint-plugin-jsx-a11y` rule `autocomplete-valid`, whose `getProp` lookup is case-insensitive and therefore does flag `autoComplete`.

### Reproduction

`test.tsx`:

```tsx
const a = <input type="text" autoComplete="foo" />; // camelCase (React) — invalid value
const b = <input type="text" autocomplete="foo" />; // lowercase (HTML)  — invalid value
export { a, b };
```

```bash
biome lint --only=a11y/useValidAutocomplete test.tsx
```

### Expected behavior

Both lines flagged — both carry the invalid autocomplete value `foo`.

### Actual behavior

Only line 2 (lowercase `autocomplete`) is flagged. Line 1 (`autoComplete`, the idiomatic React spelling) is silently ignored:

```
test.tsx:2:30 lint/a11y/useValidAutocomplete ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × Use valid values for the autocomplete attribute.

    1 │ const a = <input type="text" autoComplete="foo" />; // camelCase (React) — invalid value
  > 2 │ const b = <input type="text" autocomplete="foo" />; // lowercase (HTML)  — invalid value
      │                              ^^^^^^^^^^^^^^^^^^

Found 1 error.
```

The same applies to components registered via the `inputComponents` option: `<MyInput autoComplete="foo" />` is ignored while `<MyInput autocomplete="foo" />` is flagged.

### Likely cause

#4516 ("refactor(useValidAutocomplete): use Ast query instead of Aria") switched the rule to `JsxAttributeList::find_by_name` in place of the previous `Aria` query. The exact-match lookup appears to have dropped the case-insensitive attribute matching the rule needs to see React's `autoComplete`.

**Comment by ematipico** [comment]
Please next time use our bug report template.

Here's a playground, which wasn't submitted because OP didn't use our template: https://biomejs.dev/playground/?lintRules=useValidAutocomplete&tab=syntax&pane=Diagnostics&code=YwBvAG4AcwB0ACAAYQAgAD0AIAA8AGkAbgBwAHUAdAAgAHQAeQBwAGUAPQAiAHQAZQB4AHQAIgAgAGEAdQB0AG8AQwBvAG0AcABsAGUAdABlAD0AIgBmAG8AbwAiACAALwA%2BADsACgAKAGMAbwBuAHMAdAAgAGEAIAA9ACAAPABpAG4AcAB1AHQAIAB0AHkAcABlAD0AIgB0AGUAeAB0ACIAIABhAHUAdABvAGMAbwBtAHAAbABlAHQAZQA9ACIAZgBvAG8AIgAgAC8APgA7AA%3D%3D

**Comment by ematipico** [comment]
@hudson155 can you please provide some resource that shows that react needs `autoComplete` instead `autocomplete`?

**Comment by hudson155** [comment]
Apologies about the bug report template.

Here's a resource: https://legacy.reactjs.org/docs/dom-elements.html#all-supported-html-attributes

<img width="952" height="409" alt="Image" src="https://github.com/user-attachments/assets/a24dddfb-81c1-48bb-9e89-f81913f93bcb" />

React expects camelCase HTML attributes in general, even though pure HTML typically has them all lowercase.

**Comment by hudson155** [comment]
Here's the same page on their new (non-legacy) docs. Took me a bit longer to find it.

https://react.dev/reference/react-dom/components/input

<img width="964" height="232" alt="Image" src="https://github.com/user-attachments/assets/b2d41d40-8d8e-4fb0-94e1-9da3a23e5311" />

And more generally, https://react.dev/learn/writing-markup-with-jsx#3-camelcase-salls-most-of-the-things

**Comment by mvanhorn** [comment]
Opened #10745 to fix this. The rule now resolves the autocomplete attribute case-insensitively (`eq_ignore_ascii_case`) within `useValidAutocomplete`, so React's camelCase `autoComplete` is recognized; the shared `find_by_name` helper is left untouched. Added camelCase spec coverage and a changeset.
